cmd/fleet/main.go

```diff
@@ -846,6 +866,10 @@ func (f *FleetServer) runSubsystems(ctx context.Context, cfg *config.Config, g *
 	router := NewRouter(ctx, bulker, ct, et, at, ack, sm)
 
 	g.Go(loggedRunFunc(ctx, "Http server", func(ctx context.Context) error {
+		var router http.Handler = router
+		if f.tracer != nil {
+			router = apmhttp.Wrap(router, apmhttp.WithTracer(f.tracer))
```

Comment:
You should use https://github.com/elastic/apm-agent-go/tree/master/module/apmhttprouter so the transaction names are based on the registered route and not the request URL path.